internal/provider/data_source_aci_annotation.go

```diff
@@ -0,0 +1,107 @@
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &TagAnnotationDataSource{}
+
+func NewTagAnnotationDataSource() datasource.DataSource {
+	return &TagAnnotationDataSource{}
+}
+
+// TagAnnotationDataSource defines the data source implementation.
+type TagAnnotationDataSource struct {
+	client *client.Client
+}
+
+func (d *TagAnnotationDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	tflog.Trace(ctx, "Start metadata of datasource: aci_annotation")
+	resp.TypeName = req.ProviderTypeName + "_annotation"
+	tflog.Trace(ctx, "End metadata of datasource: aci_annotation")
+}
+
+func (d *TagAnnotationDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	tflog.Trace(ctx, "Start schema of datasource: aci_annotation")
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "The annotation datasource for the 'tagAnnotation' class",
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: "The distinguished name (DN) of the Annotation object.",
+			},
+			"parent_dn": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "The distinguished name (DN) of the parent object.",
+			},
+			"key": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: `The key or password used to uniquely identify this configuration object.`,
+			},
+			"value": schema.StringAttribute{
+				Computed:            true,
+				MarkdownDescription: `The value of the property.`,
+			},
+		},
+	}
+	tflog.Trace(ctx, "End schema of datasource: aci_annotation")
+}
+
+func (d *TagAnnotationDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	tflog.Trace(ctx, "Start configure of datasource: aci_annotation")
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		resp.Diagnostics.AddError(
+			"Provider has not been configured",
+			"The req.ProviderData is nil. Please report this issue to the provider developers.",
+		)
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	d.client = client
+	tflog.Trace(ctx, "End configure of datasource: aci_annotation")
+}
+
+func (d *TagAnnotationDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	tflog.Trace(ctx, "Start read of datasource: aci_annotation")
```

Comment:
Change the log level from trace to debug for the create, update, read and delete operations.

Reply:
ok